### PR TITLE
Feature/add nnapi tflite support

### PR DESCRIPTION
**Pull request by Edjchg**
Add support for NNAPI delegegate with TensorFlow lite backend.

### CODE REVIEW

File: r2i/nnapi/engine.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#ifndef R2I_NNAPI_H
+#define R2I_NNAPI_H
+
+#include <r2i/tflite/engine.h>
+#include "tensorflow/lite/delegates/nnapi/nnapi_delegate.h"
+#include "tensorflow/lite/tools/delegates/delegate_provider.h"
+#include "tensorflow/lite/tools/evaluation/utils.h"
```
[the remark this answers is not in the document]
Comment (lleon95) [reply r2i/nnapi/engine.h:7]: Move these headers (`tensorflow/*`) to the .cc, since they are not required here.

**Review by lleon95 — CHANGES_REQUESTED** [review]

File: meson.build [hunk 2]
```diff
@@ -51,7 +52,9 @@ if get_option('enable-tensorflow')
 endif
 
 # Define library dependencies for Tensorflow Lite support
-if get_option('enable-tflite') or get_option('enable-coral')
+if get_option('enable-tflite') or get_option('enable-coral') or get_option('enable-nnapi')
+  configinc += include_directories('../recipe-sysroot/usr/include/tensorflow/lite')
+  configinc += include_directories('../recipe-sysroot/usr/include/tensorflow/lite/tools/make/downloads/flatbuffers/include')
```
[the remark this answers is not in the document]
Comment (lleon95) [reply meson.build:57]: We could adapt it as it is shown in the R2Inference guide:
image

File: r2i/nnapi/engine.cc [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * back to RidgeRun without any encumbrance.
+ */
+
+#include "r2i/nnapi/engine.h"
+
+
+namespace r2i {
+namespace nnapi {
+Engine::Engine () : tflite::Engine() {
+  this->number_of_threads = 1;
```
[the remark this answers is not in the document]
Comment (Edjchg) [reply r2i/nnapi/engine.cc:18]: The default value greater than 0.

File: r2i/iframeworkfactory.cc [hunk 4]
```diff
@@ -81,6 +82,14 @@ MakeTensorRTFactory (RuntimeError &error) {
 }
 #endif // HAVE_TENSORRT
 
+#ifdef HAVE_NNAPI
+static std::unique_ptr<IFrameworkFactory>
+MakeNNAPIFactory (RuntimeError &error) {
+  return std::unique_ptr<nnapi::FrameworkFactory> (new
+         nnapi::FrameworkFactory);
+}
+#endif
```
[the remark this answers is not in the document]
Comment (lleon95) [reply r2i/iframeworkfactory.cc:91]: Still missing

File: r2i/nnapi/frameworkfactory.cc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+*/
+
+#include "frameworkfactory.h"
+#include "engine.h"
```
[the remark this answers is not in the document]
Comment (lleon95) [reply r2i/nnapi/frameworkfactory.cc:13]: Still missing

**Review by lleon95 — APPROVED** [review]

File: meson.build (hunk 2 above)
[the remark this answers is not in the document]
Comment (lleon95) [reply meson.build]: Please, also add the instructions over here:
https://developer.ridgerun.com/wiki/index.php?title=R2Inference/Getting_started/Building_the_library

File: r2i/iframeworkfactory.cc [hunk 6]
```diff
@@ -22,6 +22,7 @@
 #include "tensorflow/frameworkfactory.h"
 #include "tensorrt/frameworkfactory.h"
 #include "tflite/frameworkfactory.h"
+#include "nnapi/frameworkfactory.h"
```
[the remark this answers is not in the document]
Comment (lleon95) [reply r2i/iframeworkfactory.cc]: Still missing

[the remark this answers is not in the document]
Comment (lleon95) [reply r2i/iframeworkfactory.cc]: still missing

**Pull request merged**